client/my-sites/plan-features-2023-grid/hooks/npm-ready/data-store/use-grid-plans.tsx

```diff
@@ -0,0 +1,301 @@
+	| 'plans-woocommerce'
+	| 'plans-default-wpcom'
+	| 'default';
+
+interface Props {
+	// allFeaturesList temporary until feature definitions are ported to calypso-products package
+	allFeaturesList: FeatureList;
+	// API plans will be ported to data store and be queried from there
+	usePricedAPIPlans: UsePricedAPIPlans;
+	usePricingMetaForGridPlans: UsePricingMetaForGridPlans;
+	withoutProRatedCreditsForPricing?: boolean;
+	selectedFeature?: string | null;
+	term?: ( typeof TERMS_LIST )[ number ]; // defaults to monthly
+	intent?: PlansIntent;
+	selectedPlan?: PlanSlug;
+	sitePlanSlug?: PlanSlug | null;
+	hideEnterprisePlan?: boolean;
+	isInSignup?: boolean;
+	// whether plan is upgradable from current plan (used in logged-in state)
+	usePlanUpgradeabilityCheck?: ( { planSlugs }: { planSlugs: PlanSlug[] } ) => {
+		[ key: string ]: boolean;
+	};
+	showLegacyStorageFeature?: boolean;
+	// for AB Test experiment:
+	isGlobalStylesOnPersonal?: boolean;
+}
+
+const usePlanTypesWithIntent = ( {
+	intent,
+	selectedPlan,
+	sitePlanSlug,
+	hideEnterprisePlan,
+}: Pick< Props, 'intent' | 'selectedPlan' | 'sitePlanSlug' | 'hideEnterprisePlan' > ): string[] => {
+	const isEnterpriseAvailable = ! hideEnterprisePlan;
+	const isBloggerAvailable =
+		( selectedPlan && isBloggerPlan( selectedPlan ) ) ||
+		( sitePlanSlug && isBloggerPlan( sitePlanSlug ) );
+
+	let currentSitePlanType = null;
+	if ( sitePlanSlug ) {
+		currentSitePlanType = getPlan( sitePlanSlug )?.type;
+	}
+
+	const availablePlanTypes = [
+		TYPE_FREE,
+		...( isBloggerAvailable ? [ TYPE_BLOGGER ] : [] ),
+		TYPE_PERSONAL,
+		TYPE_PREMIUM,
+		TYPE_BUSINESS,
+		TYPE_ECOMMERCE,
+		...( isEnterpriseAvailable ? [ TYPE_ENTERPRISE_GRID_WPCOM ] : [] ),
+		TYPE_WOOEXPRESS_SMALL,
+		TYPE_WOOEXPRESS_MEDIUM,
+	];
+
+	let planTypes;
+	switch ( intent ) {
+		case 'plans-woocommerce':
+			planTypes = [ TYPE_WOOEXPRESS_SMALL, TYPE_WOOEXPRESS_MEDIUM ];
+			break;
+		case 'plans-blog-onboarding':
+			planTypes = [ TYPE_FREE, TYPE_PERSONAL, TYPE_PREMIUM, TYPE_BUSINESS ];
+			break;
+		case 'plans-newsletter':
+		case 'plans-link-in-bio':
+			planTypes = [ TYPE_FREE, TYPE_PERSONAL, TYPE_PREMIUM ];
+			break;
+		case 'plans-new-hosted-site':
+			planTypes = [ TYPE_FREE, TYPE_PERSONAL, TYPE_PREMIUM, TYPE_BUSINESS, TYPE_ECOMMERCE ];
+			break;
+		case 'plans-import':
+			planTypes = [ TYPE_FREE, TYPE_PERSONAL, TYPE_PREMIUM, TYPE_BUSINESS ];
+			break;
+		case 'plans-plugins':
+			planTypes = [
+				...( currentSitePlanType ? [ currentSitePlanType ] : [] ),
+				TYPE_BUSINESS,
+				TYPE_ECOMMERCE,
+			];
+			break;
+		case 'plans-jetpack-app':
+			planTypes = [ TYPE_PERSONAL, TYPE_PREMIUM ];
+			break;
+		case 'plans-default-wpcom':
+			planTypes = [
+				TYPE_FREE,
+				...( isBloggerAvailable ? [ TYPE_BLOGGER ] : [] ),
+				TYPE_PERSONAL,
+				TYPE_PREMIUM,
+				TYPE_BUSINESS,
+				TYPE_ECOMMERCE,
+				...( isEnterpriseAvailable ? [ TYPE_ENTERPRISE_GRID_WPCOM ] : [] ),
+			];
+			break;
+		default:
```

Comment:
hmm not sure - I don't think that will make things easier to code. Woo plans are not part of `plans-default-wpcom`, for instance. 

It's not a perfect structure, there's an order implied in `availablePlanTypes`, but it would need to treat special cases like Woo (or other sets not included in `plans-default-wpcom`) differently if default set doesn't include those. Might be coming back to this, but happy to keep it in a working state as-is right now.